CRAN issue with s2 code

A new notice, now concerning the s2geometry code:
```
Dear maintainer,

Please see the problems shown on
[<https://cran.r-project.org/web/checks/check_results_s2.html>](https://cran.r-project.org/web/checks/check_results_s2.html).

Please correct before 2026-05-28 to safely retain your package on CRAN.

Note that this will be the *final* reminder.

Best wishes,
The CRAN Team
```
relates to
```
Result: WARN 
  Found the following significant warnings:
    s2/mutable_s2shape_index.cc:595:31: warning: 'Lock' is deprecated [-Wdeprecated-declarations]
    s2/mutable_s2shape_index.cc:612:31: warning: 'Lock' is deprecated [-Wdeprecated-declarations]
    s2/mutable_s2shape_index.cc:640:29: warning: 'Unlock' is deprecated [-Wdeprecated-declarations]
  See ‘/home/hornik/tmp/R.check/r-devel-clang/Work/PKGS/s2.Rcheck/00install.out’ for details.
  * used C compiler: ‘Debian clang version 22.1.5 (1)’
  * used C++ compiler: ‘Debian clang version 22.1.5 (1)’
```
but also with gcc-15 and gcc-16

--- a/src/s2-altrep.cpp
+++ b/src/s2-altrep.cpp
@@ -1,5 +1,6 @@
 #define R_NO_REMAP
-#include <R.h>
+#include <Rcpp.h>
+
 #include <Rinternals.h>
 #include <Rversion.h>
 

--- a/src/s2-constructors-formatters.cpp
+++ b/src/s2-constructors-formatters.cpp
@@ -1,6 +1,6 @@
 
 #define R_NO_REMAP
-#include <R.h>
+#include <Rcpp.h>
 #include <Rinternals.h>
 
 #include "s2/s2pointutil.h"

--- a/src/s2/mutable_s2shape_index.cc
+++ b/src/s2/mutable_s2shape_index.cc
@@ -30,9 +30,20 @@
 #include "s2/base/commandlineflags.h"
 #include "s2/base/integral_types.h"
 #include "absl/base/attributes.h"
+#include "absl/base/config.h"
 #include "absl/container/btree_map.h"
 #include "absl/flags/flag.h"
 #include "absl/synchronization/mutex.h"
+
+// Abseil deprecated Lock()/Unlock() in favor of lock()/unlock() in version
+// 20260107. Use the new lowercase names when available.
+#if !defined(ABSL_LTS_RELEASE_VERSION) || ABSL_LTS_RELEASE_VERSION >= 20260107
+#define S2_ABSL_MUTEX_LOCK(mu) (mu).lock()
+#define S2_ABSL_MUTEX_UNLOCK(mu) (mu).unlock()
+#else
+#define S2_ABSL_MUTEX_LOCK(mu) (mu).Lock()
+#define S2_ABSL_MUTEX_UNLOCK(mu) (mu).Unlock()
+#endif
 #include "absl/utility/utility.h"
 #include "s2/util/coding/coder.h"
 #include "s2/util/coding/varint.h"
@@ -592,7 +603,7 @@ void MutableS2ShapeIndex::ApplyUpdatesThreadSafe() {
     // is unlocked the index_status_ is guaranteed to be FRESH.
     ++update_state_->num_waiting;
     lock_.Unlock();
-    update_state_->wait_mutex.lock();
+    S2_ABSL_MUTEX_LOCK(update_state_->wait_mutex);
     lock_.Lock();
     --update_state_->num_waiting;
     UnlockAndSignal();  // Notify other waiting threads.
@@ -609,7 +620,7 @@ void MutableS2ShapeIndex::ApplyUpdatesThreadSafe() {
     update_state_ = make_unique<UpdateState>();
     // lock_.Lock wait_mutex *before* calling Unlock() to ensure that all other
     // threads will block on it.
-    update_state_->wait_mutex.lock();
+    S2_ABSL_MUTEX_LOCK(update_state_->wait_mutex);
     // Release the spinlock before doing any real work.
     lock_.Unlock();
     ApplyUpdatesInternal();
@@ -637,7 +648,7 @@ inline void MutableS2ShapeIndex::UnlockAndSignal() {
   //
   // We need to unlock wait_mutex before destroying it even if there are no
   // waiting threads.
-  update_state_->wait_mutex.unlock();
+  S2_ABSL_MUTEX_UNLOCK(update_state_->wait_mutex);
   if (num_waiting == 0) {
     update_state_.reset();
   }

--- a/src/s2/mutable_s2shape_index.cc
+++ b/src/s2/mutable_s2shape_index.cc
@@ -592,7 +592,7 @@ void MutableS2ShapeIndex::ApplyUpdatesThreadSafe() {
     // is unlocked the index_status_ is guaranteed to be FRESH.
     ++update_state_->num_waiting;
     lock_.Unlock();
-    update_state_->wait_mutex.Lock();
+    update_state_->wait_mutex.lock();
     lock_.Lock();
     --update_state_->num_waiting;
     UnlockAndSignal();  // Notify other waiting threads.
@@ -609,7 +609,7 @@ void MutableS2ShapeIndex::ApplyUpdatesThreadSafe() {
     update_state_ = make_unique<UpdateState>();
     // lock_.Lock wait_mutex *before* calling Unlock() to ensure that all other
     // threads will block on it.
-    update_state_->wait_mutex.Lock();
+    update_state_->wait_mutex.lock();
     // Release the spinlock before doing any real work.
     lock_.Unlock();
     ApplyUpdatesInternal();
@@ -637,7 +637,7 @@ inline void MutableS2ShapeIndex::UnlockAndSignal() {
   //
   // We need to unlock wait_mutex before destroying it even if there are no
   // waiting threads.
-  update_state_->wait_mutex.Unlock();
+  update_state_->wait_mutex.unlock();
   if (num_waiting == 0) {
     update_state_.reset();
   }

--- a/src/util.cpp
+++ b/src/util.cpp
@@ -1,5 +1,5 @@
 #define R_NO_REMAP
-#include <R.h>
+#include <Rcpp.h>
 #include <Rinternals.h>
 
 #include "util.h"

--- a/src/util.h
+++ b/src/util.h
@@ -1,7 +1,7 @@
 #ifndef UTIL_H
 #define UTIL_H
 
-#include <R.h>
+#include <Rcpp.h>
 #include <Rinternals.h>
 
 extern SEXP s2_ns_pkg;

